### ISSUE TITLE
Using "sitk.Transformix()" to drive label image distortion will cause blurred boundary

### ISSUE THREAD

**Issue by oasis-github** [SuperElastix/SimpleElastix#273]
Hello!
I used “affine+bspline” parameter maps to register the images, and used "sitk.Transformix(movingLabel, selx.GetTransformParameterMap())" to drive label images deformation. But there is a problem, that the tissue boundaries in the result label image are blurred, making it impossible to distinguish which tissue these pixels belong to.

![11](https://user-images.githubusercontent.com/44250846/53616415-5936ea00-3c1d-11e9-8a16-d25fdb7a2a67.png)

It seems that some kind of smoothing is used for the label image. Can this problem be solved by parameter settings?

**Comment by kaspermarstal** [comment]
Yes, you should set "ResampleInterpolator" in the transform parameter map to "FinalNearestNeighborInterpolator" before executing transformix.

**Comment by magzHL** [comment]
I'm stuck with the same issue.
When I try to change the parameter map (according to the code fragment in: http://elastix.isi.uu.nl/marius/downloads/2016_c_WBIRa.pdf) by using:

`tp = elastixImageFilter.GetTransformParameterMap()
tp["ResampleInterpolator"] = ["FinalNearestNeighborInterpolator"]`

I get the following error:

> ---------------------------------------------------------------------------
>TypeError                                 Traceback (most recent call last)
><ipython-input-79-37276b1e15d3> in <module>
>     6 
>      7 tp = elastixImageFilter.GetTransformParameterMap()
>----> 8 tp["ResampleInterpolator"] = ["FinalNearestNeighborInterpolator"]
>      9 
>     10 resultLabel = sitk.Transformix(sitk.GetImageFromArray(npa_mov_label), tp)

>TypeError: 'tuple' object does not support item assignment

Any ideas how to solve this? I couldn't find an answer so far in the docs.

**Comment by magzHL** [comment]
Actually, I solved it right now:
`
tp = elastixImageFilter.GetTransformParameterMap() 
tp[0]["ResampleInterpolator"] = ["FinalNearestNeighborInterpolator"]
`
did the trick. As it was stated as error message, tp was a tuple...
